src/cartservice/src/cartstore/ValkeyCartStore.cs

```diff
@@ -23,6 +25,11 @@ public class ValkeyCartStore : ICartStore
     private readonly byte[] _emptyCartBytes;
     private readonly string _connectionString;
 
+    private static readonly ActivitySource CartActivitySource = new("OpenTelemetry.Demo.Cart");
+    private static readonly Meter CartMeter = new Meter("OpenTelemetry.Demo.Cart");
```

Comment:
I think that you need to listen on the traces/metrics. AddSource/AddMeter

See https://github.com/open-telemetry/opentelemetry-dotnet/blob/a6582704cd0701054f4c4cfd6d16f6925fb79144/docs/metrics/exemplars/Program.cs#L20-L25